app/services/make_an_offer.rb

```diff
@@ -0,0 +1,15 @@
+class MakeAnOffer
+  Response = Struct.new(:successful?, :application_choice)
+
+  def initialize(application_choice:, offer_conditions:)
+    @application_choice = application_choice
+    @offer_conditions = offer_conditions
+  end
+
+  def call
+    @application_choice.status = @offer_conditions.present? ? :conditional_offer : :unconditional_offer
+    @application_choice.offer = @offer_conditions.present? ? @offer_conditions : { 'conditions' => [] }
```

Comment:
We'll need to check that this is an array. I also thought we had a limit for the number of conditions, but it's [not in the docs anymore](https://apply-for-postgraduate-teacher-training-tech-docs.cloudapps.digital/reference/#request-body). Or do I misremember?

Reply:
Ah, we do but it's mentioned in the `Schema` section - https://apply-for-postgraduate-teacher-training-tech-docs.cloudapps.digital/reference/#offer.